app/content/profile/index.js

```diff
@@ -65,6 +85,147 @@ function Index(props) {
+
+    // Adjust snap points based on password input visibility
+    if (passwordInputVisible) {
+      setSnapPoints(initialSnapPoints);
+      setCurrentPassword(""); // Clear password fields
+      setNewPassword("");
+    } else {
+      setSnapPoints(expandedSnapPoints);
+    }
+  };
+
+  const handleUpdate = async () => {
+    if (name && name !== userData.name) {     // check if they request to update their name to a new one
+      await updateDoc(doc(db, "teams", "1", "users", userId), {
+        name: name,
+      });
+      bottomSheetModalRef.current.close();
+      setSnackbarMessage("Name field updated successfully");
+      setSnackbarVisible(true);     // Show success snackbar
+    }
+
+    if (email) {
+      // TODO: Decide whether we want to allow a user update their email
+    }
+
+    if (passwordInputVisible) {
+      //if password change field is open and they press update
+      if (!currentPassword && !newPassword) {
+        setDialogTitle("Error");
+        setDialogMessage("Please provide current and new passwords");
+        setDialogVisible(true);
+      } else if (!currentPassword) {
+        setDialogTitle("Error");
+        setDialogMessage("Please provide the current password");
+        setDialogVisible(true);
+      } else if (!newPassword) {
+        setDialogTitle("Error");
+        setDialogMessage("Please provide the new password");
+        setDialogVisible(true);
+      } else {
+        // attemp updating the password
+        try {
+          // re-authenticate the user and check if the provided current password is valid
+          const userCredential = await signInWithEmailAndPassword(
+            auth,
+            userEmail,
+            currentPassword,
+          );
+          // console.log("signed in:", userCredential.user);
+
+          // once re-authenticated, update the password
+          updatePassword(userCredential.user, newPassword)
+            .then(() => {
+              // Update successful
+              setCurrentPassword(""); // Clear password fields
+              setNewPassword("");
+              bottomSheetModalRef.current.close();
+              setSnackbarMessage("Password updated successfully");
+              setSnackbarVisible(true); // Show success snackbar
+            })
+            .catch((error) => {
+              // Update failed
+              console.log("password update error:", error.message);
+              setDialogTitle("New password is too short");
+              setDialogMessage("Provided new password must be at least 6 characters long!");
+              setDialogVisible(true);
+              console.log(e.message);
+            });
+        } catch (e) {
+          setDialogTitle("Error");
+          setDialogMessage("Provided current password is invalid!");
+          setDialogVisible(true);
+          console.log(e.message);
+        }
+      }
+    }
+  };
+
   if (userIsLoading || drillInfoIsLoading || attemptsIsLoading) {
```

Comment:
Move this loading and error logic as early up as possible. The only thing that cannot be below this loading screen are any funciton that start with use, as those are hooks and must be consistent between renders